packages/react/toast/src/Toast.tsx

```diff
@@ -806,6 +860,26 @@ function useNextFrame(callback = () => {}) {
   }, [fn]);
 }
 
+function getAnnounceTextContent(container: HTMLElement) {
+  const textContent = [];
+  const walker = document.createTreeWalker(container, NodeFilter.SHOW_ELEMENT, {
+    acceptNode: (node: any) => {
+      const isHidden = node.ariaHidden || node.hidden || node.style.display === 'none';
+      return isHidden ? NodeFilter.FILTER_REJECT : NodeFilter.FILTER_ACCEPT;
+    },
+  });
+
+  while (walker.nextNode()) {
+    const childNodes = Array.from(walker.currentNode.childNodes);
+    const filteredTextNodes = childNodes
+      .filter((node) => node.nodeType === node.TEXT_NODE)
+      .map((node) => node.textContent);
+    textContent.push(...filteredTextNodes);
+  }
+
+  return textContent;
+}
+
```

Comment:
Can you explain why `node.textContent` from the top-level isn't good enough? Why do we need to walk it? From what I understand `textContent` will already only return the text bits.